library/src/cache/patch_manager.rs

```diff
@@ -465,7 +465,15 @@ impl ManagePatches for PatchManager {
     }
 
     fn remove_patch(&mut self, patch_number: usize) -> Result<()> {
-        self.try_fall_back_from_patch(patch_number)
+        self.delete_patch_artifacts(patch_number)?;
+        // If the patch to remove is the next boot patch, clear it.
+        if let Some(next_boot_patch) = self.patches_state.next_boot_patch.clone() {
```

Comment:
So the change here is to clear the next_boot_patch if it happens to be the same as the patch being removed.  Do we need to clear it from other fields as well?